cmd/jsonnet.cpp

```diff
@@ -455,44 +456,53 @@ static bool process_args(int argc,
         return false;
     }
 
-    std::string filename = remaining_args[0];
-    if (remaining_args.size() > 1) {
-        std::cerr << "ERROR: Already specified " << want
-                  << " as \"" << filename << "\"\n"
-                  << std::endl;
-        usage(std::cerr);
+    bool multiple_files_allowed = config->cmd == FMT &&
+                                  (config->fmtTest || config->fmtInPlace);
+    if (!multiple_files_allowed) {
+        std::string filename = remaining_args[0];
+        if (remaining_args.size() > 1) {
+            std::cerr << "ERROR: Already specified " << want
+                      << " as \"" << filename << "\"\n"
+                      << std::endl;
+            usage(std::cerr);
+            return false;
+        }
+    }
+    config->inputFiles = remaining_args;
+    return true;
+}
+
+static bool read_input_contents(std::string filename, std::string *input) {
+    std::ifstream f;
+    f.open(filename);
+    if (!f.good()) {
+        std::string msg = "Opening input file: " + filename;
+        perror(msg.c_str());
+        return false;
+    }
+    input->assign(std::istreambuf_iterator<char>(f),
+                  std::istreambuf_iterator<char>());
+    if (!f.good()) {
+        std::string msg = "Reading input file: " + filename;
+        perror(msg.c_str());
         return false;
     }
-    config->inputFile = filename;
     return true;
 }
 
 /** Reads Jsonnet code from the input file or stdin into the input buffer. */
-static bool read_input(JsonnetConfig* config, std::string* input) {
+static bool read_input(JsonnetConfig* config, int index, std::string* input) {
     if (config->filenameIsCode) {
-        *input = config->inputFile;
-        config->inputFile = "<cmdline>";
+        *input = config->inputFiles[index];
+        config->inputFiles[index] = "<cmdline>"; // side effect!
     } else {
-        // Input file "-" tell Jsonnet to read stdin.
-        if (config->inputFile == "-") {
-            config->inputFile = "<stdin>";
+        // Input file "-" tells Jsonnet to read stdin.
+        if (config->inputFiles[index] == "-") {
```

Comment:
Looking below I guess not - this block of code reads like it could, but I think once we figure out what to do with the constant `index` parameter this won't be a problem.